packages/cli/src/CredentialTypes.ts

```diff
@@ -1,36 +1,91 @@
-import {
+import { loadClassInIsolation } from 'n8n-core';
+import type {
 	ICredentialType,
-	ICredentialTypeData,
-	ICredentialTypes as ICredentialTypesInterface,
+	ICredentialTypes,
+	INodesAndCredentials,
+	LoadedClass,
 } from 'n8n-workflow';
-import { RESPONSE_ERROR_MESSAGES } from '@/constants';
+import { RESPONSE_ERROR_MESSAGES } from './constants';
+import type { ICredentialsTypeData } from './Interfaces';
 
-class CredentialTypesClass implements ICredentialTypesInterface {
-	credentialTypes: ICredentialTypeData = {};
+class CredentialTypesClass implements ICredentialTypes {
+	constructor(private nodesAndCredentials: INodesAndCredentials) {}
 
-	async init(credentialTypes: ICredentialTypeData): Promise<void> {
-		this.credentialTypes = credentialTypes;
+	recognizes(type: string) {
+		return type in this.knownCredentials || type in this.loadedCredentials;
 	}
 
-	getAll(): ICredentialType[] {
-		return Object.values(this.credentialTypes).map((data) => data.type);
+	getByName(credentialType: string): ICredentialType {
+		return this.getCredential(credentialType).type;
 	}
 
-	getByName(credentialType: string): ICredentialType {
-		try {
-			return this.credentialTypes[credentialType].type;
-		} catch (error) {
-			throw new Error(`${RESPONSE_ERROR_MESSAGES.NO_CREDENTIAL}: ${credentialType}`);
+	/**
+	 * Returns the credentials data of the given type and its parent types it extends
+	 */
+	getCredentialsDataWithParents(type: string): ICredentialsTypeData {
```

Comment:
Comes from before, but cannot find where this is used?

Reply:
move here from `WorkflowHelpers`. buy yeah, it's unused now. will delete.